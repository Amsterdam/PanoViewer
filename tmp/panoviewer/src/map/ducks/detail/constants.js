export const REDUCER_KEY = 'mapDetail';

export const FETCH_MAP_DETAIL_REQUEST = `${REDUCER_KEY}/FETCH_MAP_DETAIL_REQUEST`;
export const FETCH_MAP_DETAIL_SUCCESS = `${REDUCER_KEY}/FETCH_MAP_DETAIL_SUCCESS`;
export const FETCH_MAP_DETAIL_FAILURE = `${REDUCER_KEY}/FETCH_MAP_DETAIL_FAILURE`;

