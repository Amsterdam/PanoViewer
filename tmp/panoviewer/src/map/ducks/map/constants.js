export const REDUCER_KEY = 'map';
// export { REDUCER_KEY as MAP };
export const MAP_BOUNDING_BOX = 'MAP_BOUNDING_BOX';
export const MAP_EMPTY_GEOMETRY = 'MAP_EMPTY_GEOMETRY';
export const MAP_END_DRAWING = 'MAP_END_DRAWING';
export const MAP_PAN = 'MAP_PAN';
export const MAP_SET_DRAWING_MODE = 'MAP_SET_DRAWING_MODE';
export const MAP_UPDATE_SHAPE = 'MAP_UPDATE_SHAPE';
export const MAP_ZOOM = 'MAP_ZOOM';
export const MAP_CLEAR = 'MAP_CLEAR';
export const SET_MAP_BASE_LAYER = 'SET_MAP_BASE_LAYER';
export const TOGGLE_MAP_OVERLAY = 'TOGGLE_MAP_OVERLAY';
export const TOGGLE_MAP_OVERLAY_PANORAMA = 'TOGGLE_MAP_OVERLAY_PANORAMA';
export const TOGGLE_MAP_OVERLAY_VISIBILITY = 'TOGGLE_MAP_OVERLAY_VISIBILITY';
export const SET_MAP_CLICK_LOCATION = 'SET_MAP_CLICK_LOCATION';
export const TOGGLE_MAP_PANEL = 'TOGGLE_MAP_PANEL';
export const CLOSE_MAP_PANEL = 'CLOSE_MAP_PANEL';
export const MAP_LOADING = 'MAP_LOADING';

export const DEFAULT_LAT = 52.3731081;
export const DEFAULT_LNG = 4.8932945;
export const PANORAMA = 'pano';

export const initialState = {
  viewCenter: [DEFAULT_LAT, DEFAULT_LNG],
  baseLayer: 'topografie',
  zoom: 11,
  overlays: [],
  isLoading: false,
  drawingMode: 'none',
  shapeMarkers: 0,
  shapeDistanceTxt: '',
  shapeAreaTxt: '',
  mapPanelActive: false
};

