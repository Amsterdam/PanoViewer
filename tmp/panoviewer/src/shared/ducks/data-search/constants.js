export const REDUCER_KEY = 'dataSearch';
export const FETCH_GEO_SEARCH_RESULTS_REQUEST = `${REDUCER_KEY}/FETCH_GEO_SEARCH_RESULTS_REQUEST`;
export const FETCH_GEO_SEARCH_RESULTS_FAILURE = `${REDUCER_KEY}/FETCH_GEO_SEARCH_RESULTS_FAILURE`;
export const FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL = `${REDUCER_KEY}/FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL`;
export const FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST = `${REDUCER_KEY}/FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST`;

export const FETCH_QUERY_SEARCH_RESULTS_REQUEST = `${REDUCER_KEY}/FETCH_QUERY_SEARCH_RESULTS_REQUEST`;
export const FETCH_QUERY_SEARCH_RESULTS_SUCCESS = `${REDUCER_KEY}/FETCH_QUERY_SEARCH_RESULTS_SUCCESS`;
export const FETCH_QUERY_SEARCH_RESULTS_FAILURE = `${REDUCER_KEY}/FETCH_QUERY_SEARCH_RESULTS_FAILURE`;

export const FETCH_QUERY_SEARCH_MORE_RESULTS_REQUEST = `${REDUCER_KEY}/FETCH_QUERY_SEARCH_MORE_RESULTS_REQUEST`;
export const FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS = `${REDUCER_KEY}/FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS`;
export const FETCH_QUERY_SEARCH_MORE_RESULTS_FAILURE = `${REDUCER_KEY}/FETCH_QUERY_SEARCH_MORE_RESULTS_FAILURE`;

export const REQUEST_NEAREST_DETAILS = `${REDUCER_KEY}/REQUEST_NEAREST_DETAILS`;

export const SET_QUERY_CATEGORY = `${REDUCER_KEY}/SET_QUERY_CATEGORY`;

export const initialState = {
  query: '',
  resultsMapPanel: [],
  resultsMap: [],
  resultsQuery: [],
  isLoading: false,
  numberOfResults: 0,
  category: null,
  error: null
};

