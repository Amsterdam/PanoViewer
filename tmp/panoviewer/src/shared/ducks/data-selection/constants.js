import { routing } from '../../../app/routes';
import { VIEW_MODE } from '../ui/ui';

export const REDUCER_KEY = 'dataSelection';
export const FETCH_DATA_SELECTION_REQUEST = `${REDUCER_KEY}/FETCH_DATA_SELECTION_REQUEST`;
export const FETCH_DATA_SELECTION_SUCCESS = `${REDUCER_KEY}/FETCH_DATA_SELECTION_SUCCESS`;
export const FETCH_DATA_SELECTION_FAILURE = `${REDUCER_KEY}/FETCH_DATA_SELECTION_FAILURE`;

export const FETCH_MARKERS_REQUEST = `${REDUCER_KEY}/FETCH_MARKERS_REQUEST`;
export const FETCH_MARKERS_SUCCESS = `${REDUCER_KEY}/FETCH_MARKERS_SUCCESS`;
export const FETCH_MARKERS_FAILURE = `${REDUCER_KEY}/FETCH_MARKERS_FAILURE`;

export const DOWNLOAD_DATA_SELECTION = `${REDUCER_KEY}/DOWNLOAD_DATA_SELECTION`;
export const REMOVE_GEOMETRY_FILTER = `${REDUCER_KEY}/REMOVE_GEOMETRY_FILTER`;
export const SET_GEOMETRY_FILTER = `${REDUCER_KEY}/SET_GEOMETRY_FILTER`;
export const RESET_DATA_SELECTION = `${REDUCER_KEY}/RESET_DATA_SELECTION`;
export const CANCEL_DATA_SELECTION = `${REDUCER_KEY}/CANCEL_DATA_SELECTION`;
export const START_DATA_SELECTION = `${REDUCER_KEY}/START_DATA_SELECTION`;
export const END_DATA_SELECTION = `${REDUCER_KEY}/END_DATA_SELECTION`;

export const SET_PAGE = `${REDUCER_KEY}/SET_PAGE`;
export const SET_DATASET = `${REDUCER_KEY}/SET_DATASET`;
export const ROUTE_DATASET_MAPPER = {
  [routing.cadastralObjects.type]: 'brk',
  [routing.establishments.type]: 'hr',
  [routing.addresses.type]: 'bag'
};

export const VIEWS_TO_PARAMS = {
  [VIEW_MODE.SPLIT]: 'LIST',
  [VIEW_MODE.MAP]: 'MAP',
  [VIEW_MODE.FULL]: 'TABLE'
};

export const DATASETS = {
  BAG: 'bag',
  BRK: 'brk',
  HR: 'hr'
};

export const DATASET_ROUTE_MAPPER = {
  [DATASETS.HR]: routing.establishments.type,
  [DATASETS.BAG]: routing.addresses.type,
  [DATASETS.BRK]: routing.cadastralObjects.type
};

export const initialState = {
  isLoading: false,
  loadingMarkers: false,
  markers: [], // eg: [[52.1, 4.1], [52.2, 4.0]],
  geometryFilter: {
    markers: undefined
  },
  dataset: DATASETS.BAG,
  authError: false,
  errorMessage: '',
  page: 1
};
