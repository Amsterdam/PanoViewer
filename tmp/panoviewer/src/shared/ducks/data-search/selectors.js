import { createSelector } from 'reselect';
import { REDUCER_KEY } from './constants';

export const getDataSearch = (state) => state[REDUCER_KEY];
const getStateOfKey = (key) =>
  (state) => createSelector(getDataSearch, (data) => (data[key]))(state);

// Data to search for
export const getDataSearchLocation = getStateOfKey('geoSearch');
export const isSearchLoading = getStateOfKey('isLoading');
export const getSearchQuery = getStateOfKey('query');
export const getSearchCategory = getStateOfKey('category');

// Results
export const getSearchQueryResults = getStateOfKey('resultsQuery');
export const getMapPanelResults = getStateOfKey('resultsMapPanel');
export const getMapListResults = getStateOfKey('resultsMap');

// Misc
export const getDataSearchError = getStateOfKey('error');
export const getNumberOfResults = getStateOfKey('numberOfResults');
