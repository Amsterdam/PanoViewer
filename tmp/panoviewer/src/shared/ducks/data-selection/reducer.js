import PAGES from '../../../app/pages';
import { shouldResetState } from '../../../store/redux-first-router/actions';
import {
  FETCH_DATA_SELECTION_FAILURE,
  FETCH_DATA_SELECTION_REQUEST,
  FETCH_DATA_SELECTION_SUCCESS,
  FETCH_MARKERS_FAILURE,
  FETCH_MARKERS_REQUEST,
  FETCH_MARKERS_SUCCESS,
  initialState,
  REDUCER_KEY,
  REMOVE_GEOMETRY_FILTER,
  RESET_DATA_SELECTION,
  ROUTE_DATASET_MAPPER,
  SET_DATASET,
  SET_GEOMETRY_FILTER,
  SET_PAGE
} from './constants';
import { routing } from '../../../app/routes';
import { SET_SELECTION } from '../selection/selection';
import { FETCH_MAP_DETAIL_SUCCESS } from '../../../map/ducks/detail/constants';
import paramsRegistry from '../../../store/params-registry';

export { REDUCER_KEY as DATA_SELECTION };

export default function reducer(state = initialState, action) {
  if (shouldResetState(action, [PAGES.ADDRESSES, PAGES.ESTABLISHMENTS, PAGES.CADASTRAL_OBJECTS])) {
    return initialState;
  }
  const enrichedState = {
    ...state,
    ...paramsRegistry.getStateFromQueries(REDUCER_KEY, action),
    ...(ROUTE_DATASET_MAPPER[action.type]) ? { dataset: ROUTE_DATASET_MAPPER[action.type] } : {}
  };

  switch (action.type) {
    case routing.home.type:
    case FETCH_MAP_DETAIL_SUCCESS:
    case SET_SELECTION: {
      return {
        ...initialState
      };
    }

    case FETCH_MARKERS_REQUEST:
      return {
        ...enrichedState,
        loadingMarkers: true
      };

    case FETCH_DATA_SELECTION_REQUEST:
      return {
        ...enrichedState,
        dataset: action.payload.dataset,
        page: action.payload.page,
        isLoading: true
      };

    case FETCH_DATA_SELECTION_SUCCESS: {
      return {
        ...enrichedState,
        isLoading: false,
        markers: [],
        errorMessage: '',
        authError: false,
        ...action.payload
      };
    }

    case FETCH_DATA_SELECTION_FAILURE:
      return {
        ...enrichedState,
        isLoading: false,
        authError: (action.payload === 'Unauthorized'),
        errorMessage: action.payload,
        result: {},
        markers: []
      };

    case FETCH_MARKERS_FAILURE:
      return {
        ...enrichedState,
        loadingMarkers: false,
        errorMessage: action.payload,
        result: {},
        markers: []
      };

    case FETCH_MARKERS_SUCCESS:
      return {
        ...enrichedState,
        loadingMarkers: false,
        markers: action.payload
      };

    case SET_DATASET:
      return {
        ...enrichedState,
        dataset: action.payload
      };

    case SET_GEOMETRY_FILTER: {
      const { markers, distanceTxt, areaTxt } = action.payload;
      return {
        ...enrichedState,
        geometryFilter: {
          markers,
          description: `${distanceTxt} en ${areaTxt}`
        }
      };
    }

    case REMOVE_GEOMETRY_FILTER:
      return {
        ...enrichedState,
        geometryFilter: {}
      };

    case SET_PAGE:
      return {
        ...enrichedState,
        page: action.payload
      };

    case RESET_DATA_SELECTION:
      return {
        ...enrichedState,
        shape: ''
      };

    default:
      return enrichedState;
  }
}
