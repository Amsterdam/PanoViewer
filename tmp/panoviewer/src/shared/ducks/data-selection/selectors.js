// Selectors
import get from 'lodash.get';
import { createSelector } from 'reselect';
import { REDUCER_KEY } from './constants';
import { detailPointType } from '../../../map/components/leaflet/services/icons.constant';

export const getDataSelection = (state) => state[REDUCER_KEY];
export const getDataSelectionPage = createSelector(
  getDataSelection,
  (dataSelection) => dataSelection.page);

export const getGeometryFilter = createSelector(
  getDataSelection,
  (dataSelection) => dataSelection.geometryFilter);
export const getGeometryFiltersMarkers = createSelector(
  getGeometryFilter,
  (filters) => (filters && filters.markers) || []);

export const getDataset = createSelector(
  getDataSelection,
  (dataSelection) => dataSelection.dataset);

export const getDataSelectionResult = createSelector(
  getDataSelection,
  (dataSelection) => dataSelection.result || {});
const generateMarkers = (markers) => (
  markers.map((markerLocation, index) => ({
    position: markerLocation,
    type: detailPointType,
    index
  })));
const getMapMarkers = createSelector([getDataSelection],
  (dataSelection) => dataSelection.markers || []);

export const getClusterMarkers = createSelector([getMapMarkers],
  (markers) => (
    markers && markers.clusterMarkers && markers.clusterMarkers.length ?
      generateMarkers(markers.clusterMarkers) : []
  ));

export const getBrkMarkers = createSelector([getMapMarkers],
  (markers) => (
    markers && markers.markers && markers.markers.length ?
      markers.markers : []
  ));
export const getGeoJsons = createSelector([getMapMarkers],
  (markers) => (
    markers && markers.geoJsons && markers.geoJsons.length ?
      markers.geoJsons : []
  ));

export const getShapeFilter = createSelector(getDataSelection, (dataSelection) => {
  const markers = get(dataSelection, 'geometryFilter.markers');
  const description = get(dataSelection, 'geometryFilter.description');
  return (markers && description) ? {
    shape: {
      slug: 'shape',
      label: 'Locatie',
      option: `ingetekend (${description})`
    }
  } : {};
});

export const getFilters = createSelector(
  getDataSelectionResult, (result) => result.filters || []
);
export const getGeomarkersShape = createSelector(
  getGeometryFiltersMarkers,
  (markers) => JSON.stringify(
    markers.map(([lat, lng]) => [lng, lat])
  )
);
export const areMarkersLoading = createSelector(
  getDataSelection,
  (dataSelection) => dataSelection.loadingMarkers
);
