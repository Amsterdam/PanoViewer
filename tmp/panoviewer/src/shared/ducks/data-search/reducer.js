import {
  FETCH_GEO_SEARCH_RESULTS_FAILURE,
  FETCH_GEO_SEARCH_RESULTS_REQUEST,
  FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST,
  FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL,
  FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS,
  FETCH_QUERY_SEARCH_RESULTS_FAILURE,
  FETCH_QUERY_SEARCH_RESULTS_REQUEST,
  FETCH_QUERY_SEARCH_RESULTS_SUCCESS,
  initialState,
  REDUCER_KEY,
  SET_QUERY_CATEGORY
} from './constants';
import { routing } from '../../../app/routes';
import { FETCH_DATA_SELECTION_REQUEST } from '../data-selection/constants';
import paramsRegistry from '../../../store/params-registry';

export { REDUCER_KEY as DATA_SEARCH_REDUCER };

export default function reducer(state = initialState, action) {
  const enrichedState = {
    ...state,
    ...paramsRegistry.getStateFromQueries(REDUCER_KEY, action)
  };

  switch (action.type) {
    case routing.dataSearchCategory.type:
      return {
        ...enrichedState,
        category: action.payload.category
      };

    case FETCH_QUERY_SEARCH_RESULTS_REQUEST:
      return {
        ...enrichedState,
        isLoading: true,
        resultsQuery: [],
        numberOfResults: 0,
        query: action.payload
      };

    case FETCH_QUERY_SEARCH_RESULTS_SUCCESS: {
      const { results, numberOfResults } = action.payload;
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults,
        resultsQuery: results
      };
    }

    case FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS: {
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults: 999,
        resultsQuery: [{
          ...state.resultsQuery[0],
          ...action.payload[0]
        }]
      };
    }

    case FETCH_GEO_SEARCH_RESULTS_REQUEST:
      return {
        ...enrichedState,
        isLoading: true,
        geoSearch: action.payload
      };

    case FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL: {
      const { results, numberOfResults } = action.payload;
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults,
        resultsMapPanel: results
      };
    }

    case FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST: {
      const { results, numberOfResults } = action.payload;
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults,
        resultsMap: results
      };
    }

    case FETCH_QUERY_SEARCH_RESULTS_FAILURE:
    case FETCH_GEO_SEARCH_RESULTS_FAILURE:
      return {
        ...enrichedState,
        isLoading: false,
        error: action.payload
      };

    case SET_QUERY_CATEGORY:
      return {
        ...enrichedState,
        category: action.payload
      };

    case FETCH_DATA_SELECTION_REQUEST:
      return {
        ...enrichedState
      };

    default:
      return enrichedState;
  }
}
