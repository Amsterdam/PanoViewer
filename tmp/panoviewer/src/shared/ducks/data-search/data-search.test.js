import {
  fetchMapSearchResultsFailure,
  fetchMapSearchResultsRequest,
  fetchMapSearchResultsSuccessPanel
} from './actions';
import { REDUCER_KEY } from './constants';
import reducer from './reducer';
import { getMapPanelResults, getSearchQuery } from './selectors';

describe('mapSearch reducer', () => {
  it('sets the initial state', () => {
    expect(reducer(undefined, { type: 'UNKOWN' })).toMatchSnapshot();
  });

  it('handles search results request', () => {
    const action = fetchMapSearchResultsRequest({
      latitude: 52.3637006,
      longitude: 4.7943446
    });
    expect(reducer({}, action)).toMatchSnapshot();
  });

  it('handles search results success', () => {
    const action = fetchMapSearchResultsSuccessPanel([{ foo: 'bar' }]);
    expect(reducer({}, action)).toMatchSnapshot();
  });

  it('should handle search failure', () => {
    expect(reducer({}, fetchMapSearchResultsFailure('error'))).toEqual({
      isLoading: false,
      error: 'error'
    });
  });
});

describe('data-search selectors', () => {
  it('getSearchQuery', () => {
    const state = {
      [REDUCER_KEY]: {
        query: 'foo'
      }
    };
    expect(getSearchQuery(state)).toEqual(state[REDUCER_KEY].query);
  });

  describe('getMapPanelResults', () => {
    it('should return state.resultsMap as a array', () => {
      const state = {
        [REDUCER_KEY]: {
          resultsMapPanel: []
        }
      };
      expect(getMapPanelResults(state)).toEqual(state[REDUCER_KEY].resultsMapPanel);
    });
  });
});
