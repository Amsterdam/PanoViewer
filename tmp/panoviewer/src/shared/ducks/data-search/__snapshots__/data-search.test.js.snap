// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapSearch reducer handles search results request 1`] = `
Object {
  "geoSearch": Object {
    "latitude": 52.3637006,
    "longitude": 4.7943446,
  },
  "isLoading": true,
}
`;

exports[`mapSearch reducer handles search results success 1`] = `
Object {
  "isLoading": false,
  "numberOfResults": undefined,
  "resultsMapPanel": Array [
    Object {
      "foo": "bar",
    },
  ],
}
`;

exports[`mapSearch reducer sets the initial state 1`] = `
Object {
  "category": null,
  "error": null,
  "isLoading": false,
  "numberOfResults": 0,
  "query": "",
  "resultsMap": Array [],
  "resultsMapPanel": Array [],
  "resultsQuery": Array [],
}
`;
